runner/server/helpers.go

```diff
@@ -0,0 +1,69 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/spaceuptech/helpers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/spaceuptech/space-cloud/runner/model"
+)
+
+func generateLogRequestFromQueryParams(ctx context.Context, r *url.URL) (*model.LogRequest, error) {
+	taskID := r.Query().Get("taskId")
+	replicaID := r.Query().Get("replicaId")
+	since := r.Query().Get("since")
+	sinceTime := r.Query().Get("since-time")
+	tail := r.Query().Get("tail")
+	_, isFollow := r.Query()["follow"]
+
+	req := &model.LogRequest{
+		TaskID:    taskID,
+		ReplicaID: replicaID,
+		IsFollow:  isFollow,
+	}
+
+	if replicaID == "" {
+		return nil, helpers.Logger.LogError(helpers.GetRequestID(ctx), "replica id not provided in query param", nil, map[string]interface{}{"since": since, "sinceTime": sinceTime})
+	}
+
+	if since != "" && sinceTime != "" {
+		return nil, helpers.Logger.LogError(helpers.GetRequestID(ctx), "you cannot only provide since or sinceTime at once", nil, map[string]interface{}{"since": since, "sinceTime": sinceTime})
+	}
+
+	if since != "" {
+		sinceNum, err := time.ParseDuration(since)
+		if err != nil {
+			return nil, helpers.Logger.LogError(helpers.GetRequestID(ctx), fmt.Sprintf("cannot parse since value (%s) to time", since), err, map[string]interface{}{"since": since})
```

Comment:
```suggestion
			return nil, helpers.Logger.LogError(helpers.GetRequestID(ctx), fmt.Sprintf("Cannot parse since value (%s) to time", since), err, map[string]interface{}{"since": since})
```